source/locationHelper.py

```diff
@@ -0,0 +1,398 @@
+#locationHelper.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited, Babbage B.V.
+
+"""Classes and helper functions for working with rectangles and coordinates."""
+
+from collections import namedtuple
+import windowUtils
+import winUser
+from ctypes.wintypes import RECT, SMALL_RECT, POINT
+import textInfos
+import wx
+
+class Point(namedtuple("Point",("x","y"))):
+	"""Represents a point on the screen."""
+
+	def __add__(self,other):
+		"""Returns a new L{Point} with its coordinates representing the additions of the original x and y coordinates."""
+		if not isinstance(other,POINT_CLASSES):
+			return NotImplemented
+		return Point((self.x+other.x),(self.y+other.y))
+
+	def __radd__(self,other):
+		"""Returns a new L{Point} with x = self.x + other.x and y = self.y + other.y."""
+		return self.__add__(other)
+
+	def __sub__(self,other):
+		if not isinstance(other,POINT_CLASSES):
+			return NotImplemented
+		return Point((self.x-other.x),(self.y-other.y))
+
+	def __rsub__(self,other):
+		return self.__sub__(other)
+
+	def __lt__(self,other):
+		"""
+		Returns whether self is less than other.
+		This first compares y, then x coordinates.

```

Comment:
p = How about Point(1,2)
p.compare(Point.Y_FIRST)
or p.compare(Point.Y_THEN_X)

Reply:
Hmm, I'm stull a bit puzzled about what names to use. xWiseLessThan sounds ok, but what to use for the le operator? xWiseLessThanOrEqual is a bit long. How about?

* yWiseLE
* yWiseLT
* yWiseGE
* yWiseGT

Etc.

@derekriemer: how would less than or equal to fit into your compare idea? Similar to the cmp method, returning 1, 0 or -1? It would be the simplest solution, not very pythonic though IMO.

Reply:
You could shorten to:
- `xWiseLessOrEq`
- `xWiseLessThan`
- `xWiseGreaterThan`
- `xWiseGreaterOrEq`

I think that is a nice balance between brevity and clarity.
I would prefer not to return 1, 0, -1 from a single compare function.
Another option (borrowing from Derek's idea) would be:
`point1.LessThan(Point.xWise, point2)` 
Though this is longer, with no more clarity. Though probably less lines of code in the Point class required.

Reply:
I prefer your first suggestion.